spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -1091,21 +1094,9 @@ private ExecutorService getService() {
 
   /**
    * Returns the first partition spec in an IcebergTable that shares the same names and ordering as
-   * the partition columns in a given Spark Table. Throws an error if not found
+   * the partition columns provided. Throws an error if not found
    */
-  private static PartitionSpec findCompatibleSpec(
-      Table icebergTable, SparkSession spark, String sparkTable) throws AnalysisException {
-    List<String> parts = Lists.newArrayList(Splitter.on('.').limit(2).split(sparkTable));
-    String db = parts.size() == 1 ? "default" : parts.get(0);
-    String table = parts.get(parts.size() == 1 ? 0 : 1);
-
-    List<String> sparkPartNames =
-        spark.catalog().listColumns(db, table).collectAsList().stream()
-            .filter(org.apache.spark.sql.catalog.Column::isPartition)
-            .map(org.apache.spark.sql.catalog.Column::name)
-            .map(name -> name.toLowerCase(Locale.ROOT))
-            .collect(Collectors.toList());
-
+  public static PartitionSpec findCompatibleSpec(List<String> partitionNames, Table icebergTable) {
```

Comment:
Could we keep this package private? Just to avoid exposing another api to the general public unless we have a need to.